### PR TITLE
Add reason to session state did change handler

### LINKED ISSUE #10: No session state change reason yet

### PR DESCRIPTION
**Pull request by nicolaslee2775**
based on PR #18
fix #10

### CODE REVIEW

File: Sources/Authgear.swift [hunk 1]
```diff
@@ -531,15 +539,15 @@ public class Authgear: NSObject {
...
+                handler?(self.persistSession(tokenResponse, reason: .foundToken))
             } catch {
                 if let error = error as? AuthAPIClientError,
-                    case let .oidcError(oidcError) = error,
-                    oidcError.error == "invalid_grant" {
+                   case let .oidcError(oidcError) = error,
+                   oidcError.error == "invalid_grant" {
                     return DispatchQueue.main.async {
-                        self.setSessionState(.noSession)
+                        self.setSessionState(.noSession, reason: .expired)
```
Comment (nicolaslee2775) [inline Sources/Authgear.swift:548]: Should we remove line 548?
`authgearSessionStateDidChange` would be called twice.

**Comment by nicolaslee2775** [comment]
@louischan-oursky Rebased.

**Comment by louischan-oursky** [comment]
conflicted.

**Comment by nicolaslee2775** [comment]
@louischan-oursky Rebased.

**Comment by louischan-oursky** [comment]
again.

**Comment by nicolaslee2775** [comment]
Rebased

**Pull request merged**